myia/prim/type_inferrers.py

```diff
@@ -199,6 +199,75 @@ def default(self):
     return t
 
 
+@type_inferrer(P.shape, nargs=1)
+async def infer_type_shape(engine, ary):
+    """Infer the return type of shape."""
+    shp = await engine.get('shape', ary)
+    return Tuple([Int(64)]*len(shp))
```

Comment:
Is `shp` guaranteed to be a tuple? Just making sure.

Reply:
Yes